packages/babel-plugin-transform-react-constant-elements/src/index.js

```diff
@@ -53,6 +55,19 @@ export default function ({ types: t }) {
         if (path.node._hoisted) return;
 
         const state = { isImmutable: true };
+
+        // This transform takes the option `allowMutablePropsOnTags`, which is an array
+        // of JSX tags to allow mutable props (such as objects, functions) on. Use sparingly
+        // and only on tags you know will never modify their own props.
+        if (this.opts.allowMutablePropsOnTags != null) {
+          if (!Array.isArray(this.opts.allowMutablePropsOnTags)) {
+            throw new Error(".allowMutablePropsOnTags must be an array, null, or undefined.");
+          }
+          const elementName = path.node.openingElement.name.name;
```

Comment:
We'd want the property, not the object, no? Or would we want to join the whole expression?

At the moment I have:

```js
// Get the element's name. If it's a member expression, we use the last part of the path.
// So the option ["FormattedMessage"] would match "Intl.FormattedMessage".
let namePath = path.get("openingElement.name");
while (namePath.isJSXMemberExpression()) namePath = namePath.get("property");

const elementName = namePath.node.name;
state.mutablePropsAllowed = this.opts.allowMutablePropsOnTags.includes(elementName);
```